src/System/Analytics/AnalyticsContext.tsx

```diff
@@ -1,20 +1,163 @@
-import { createContext, useContext } from "react"
-import { PageOwnerType } from "@artsy/cohesion"
+import {
+  FC,
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react"
+import { OwnerType, PageOwnerType } from "@artsy/cohesion"
+import { getENV } from "Utils/getENV"
+import { useToasts, useUpdateEffect } from "@artsy/palette"
+import { camelCase } from "lodash"
+import { useRouter } from "System/Router/useRouter"
 
 export interface AnalyticsContextProps {
   contextPageOwnerId?: string
   contextPageOwnerSlug?: string
   contextPageOwnerType?: PageOwnerType
 }
 
-export const AnalyticsContext = createContext<{
+const AnalyticsContext = createContext<{
   contextPageOwnerId?: string
   contextPageOwnerSlug?: string
-  contextPageOwnerType?: PageOwnerType
-  children?: any
-}>({})
+  contextPageOwnerType: PageOwnerType
+  isReady: boolean
+  isProvided: boolean
+}>({
+  contextPageOwnerType: (undefined as any) as PageOwnerType,
+  isReady: false,
+  isProvided: false,
+})
 
 export const useAnalyticsContext = () => {
-  const analyticsContext = useContext(AnalyticsContext)
-  return analyticsContext
+  const context = useContext(AnalyticsContext)
+
+  const { sendToast } = useToasts()
+  const { match } = useRouter()
+
+  // Warn if usage is attempted without providing a context
+  useEffect(() => {
+    if (context.isProvided) return
+
+    sendToast({
+      message: "Please wrap your app in `AnalyticsContextProvider`",
+      ttl: Infinity,
+      variant: "error",
+    })
+  }, [context, sendToast, match])
+
+  return context
```

Comment:
Is this duplicated (with the development env check) [down below](https://github.com/artsy/force/pull/12747/files#diff-91d000540513622973ef9d699e2cfd5ae4444c21ef601128c970314655d42f5eR111-R125)?

Reply:
This one warns if you attempt to use the context without providing it. The one below warns you if we can't infer the `contextPageOwnerType`

Reply:
And actually; This definitely should be guarded with a dev check as well.